pkg/virt-controller/watch/migration.go

```diff
@@ -1021,10 +1057,30 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 		pod = pods[0]
 	}
 
-	vmiDeleted := vmi == nil || vmi.DeletionTimestamp != nil
-	migrationFinalizedOnVMI := vmi.Status.MigrationState != nil && vmi.Status.MigrationState.MigrationUID == migration.UID && vmi.Status.MigrationState.EndTimestamp != nil
+	if vmiDeleted := vmi == nil || vmi.DeletionTimestamp != nil; vmiDeleted {
+		return nil
+	}
+
+	if (migration.DeletionTimestamp != nil && !migration.IsFinal()) || c.isMigrationAborted(migration.UID) {
+		err := c.handleSignalMigrationAbort(migration, vmi)
+		if err != nil {
+			return err
+		}
+
+		if migration.Status.Phase != virtv1.MigrationRunning && targetPodExists && pod.Status.Phase == k8sv1.PodPending {
+			// In this case when the migration has been aborted and target pod is still not ready and migration did
+			// not start to run, delete target pod
+			err = c.clientset.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, v1.DeleteOptions{})
```

Comment:
you shouldn't need to do this here.

there's a handlePendingPodTimeout() function. If the migration has been aborted, short circuit that handlePendingPodTimeout function to delete the pod early.

Reply:
As far as I can tell, `handlePendingPodTimeout()` won't help us with this specific case.

As the comment says, this is meant to deal with a scenario where the migration has been aborted and target pod is still not ready. Since this is the case, the `secondsSpentPending` (in `handlePendingPodTimeout()` function) will be set to 0, therefore no pod would be deleted.

What I can do, instead, is call `deleteTimedOutTargetPod()`. This function goes straight into raising an expectation for the pod's deletion and aims to delete it. It's also a bit confusing, since there the pod is not "timed out" but aborted, but we can maybe change the function name to `deleteTargetPod()` to make this clearer.

Am I missing something here? Does this make sense to you?